File: schrottverwaltung/src/main/java/de/fhdortmund/schrottverwaltung/bewertung/Kriterium.java
package de.fhdortmund.schrottverwaltung.bewertung;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Table(name = "kriterium")
@Getter
@Setter
public class Kriterium {
    @Column(name = "bezeichnung")
    private String bezeichnung;

    @Column(name = "max_gewichtung")
    private int maxGewichtung;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "kriterium_id")
    private Long id;

    public Kriterium(String bezeichnung, int maxGewichtung) {
        this.bezeichnung = bezeichnung;
        this.maxGewichtung = maxGewichtung;
    }
}

File: schrottverwaltung/src/main/kotlin/de/fhdortmund/schrottverwaltung/bewertung/Kriterium.kt
package de.fhdortmund.schrottverwaltung.bewertung

import jakarta.persistence.*

@Entity
@Table(name = "kriterium")
data class Kriterium(
    @Column(name = "bezeichnung")
    var bezeichnung: String,
    @Column(name = "max_gewichtung")
    var maxGewichtung: Int,
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "kriterium_id")
    var id: Long? = null
)
